### PR TITLE
Fix non-fast-start files demuxer bug

### PR DESCRIPTION
**Pull request by Noarkhh**
- Fix a bug where a demuxer trying to demux a non-fast-start file with option `optimize_for_non_fast_start?` set to `false` would crash
- Add tests covering that case

### CODE REVIEW

File: examples/demuxer_isom.exs [hunk 1]
```diff
@@ -39,7 +39,7 @@ defmodule Example do
     state = %{state | children_with_eos: MapSet.put(state.children_with_eos, element)}
 
     actions =
-      if Enum.all?([:sink_video, :sink_audio], &(&1 in state.children_with_eos)),
+      if Enum.all?([:sink_audio], &(&1 in state.children_with_eos)),
```
Comment (mat-hek) [inline examples/demuxer_isom.exs:42]: why this?

File: lib/membrane_mp4/track.ex [hunk 2]
```diff
@@ -67,21 +67,19 @@ defmodule Membrane.MP4.Track do
...
-      {:mp4a, map}
-    end
+    map = %{
+      aot_id: AAC.profile_to_aot_id(profile),
+      channels: AAC.channels_to_channel_config_id(channels),
+      frequency: sample_rate
+    }
+
+    {:mp4a, map}
+    # end
```
Comment (mat-hek) [inline lib/membrane_mp4/track.ex:82]: leftover

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: examples/demuxer_isom.exs [hunk 3]
```diff
@@ -18,7 +18,7 @@ defmodule Example do
         location: @input_file,
         hackney_opts: [follow_redirect: true]
       })
-      |> child(:demuxer, %Membrane.MP4.Demuxer.ISOM{optimize_for_non_fast_start?: true})
+      |> child(:demuxer, Membrane.MP4.Demuxer.ISOM)
```
[the remark this answers is not in the document]
Comment (Noarkhh) [reply examples/demuxer_isom.exs:21]: it's not, and the bug was occuring for non-fast-start files with `optimize_for_non_fast_start?: false`

File: examples/demuxer_isom.exs (hunk 1 above)
Comment (Noarkhh) [reply examples/demuxer_isom.exs:42]: my oversight, was testing for audio only files and forgot to revert

**Pull request merged**